packages/pluginmanager/src/model.ts

```diff
@@ -0,0 +1,474 @@
+      });
+      if (!userConfirmation.button.accept) {
+        return;
+      }
+    }
+
+    await this._performAction('disable', entry);
+    if (this.actionError) {
+      return;
+    }
+    entry.enabled = false;
+  }
+
+  protected getDependants(entry: IEntry): {
+    dependants: IEntry[];
+    optionalDependants: IEntry[];
+  } {
+    const dependants = [];
+    const optionalDependants = [];
+    if (entry.provides) {
+      const tokenName = entry.provides.name;
+      for (const plugin of this._available.values()) {
+        if (!plugin.enabled) {
+          continue;
+        }
+        if (
+          plugin.requires
+            .filter(token => !!token)
+            .some(token => token.name === tokenName)
+        ) {
+          dependants.push(plugin);
+        }
+        if (
+          plugin.optional
+            .filter(token => !!token)
+            .some(token => token.name === tokenName)
+        ) {
+          optionalDependants.push(plugin);
+        }
+      }
+    }
+    return {
+      dependants,
+      optionalDependants
+    };
+  }
+
+  /**
+   * Whether there are currently any actions pending.
+   */
+  hasPendingActions(): boolean {
+    return this._pendingActions.length > 0;
+  }
+
+  /**
+   * Send a request to the server to perform an action on a plugin.
+   *
+   * @param action A valid action to perform.
+   * @param entry The plugin to perform the action on.
+   */
+  private _performAction(action: string, entry: IEntry): Promise<IActionReply> {
+    const actionRequest = this._requestAPI<IActionReply>(
+      {},
+      {
+        method: 'POST',
+        body: JSON.stringify({
+          cmd: action,
+          plugin_name: entry.id
+        })
+      }
+    );
+
+    actionRequest.then(
+      _ => {
+        this.actionError = null;
+      },
+      reason => {
+        this.actionError = reason.toString();
+      }
+    );
+    this._addPendingAction(actionRequest);
+    return actionRequest;
+  }
+
+  /**
+   * Add a pending action.
+   *
+   * @param pending A promise that resolves when the action is completed.
+   */
+  private _addPendingAction(pending: Promise<any>): void {
+    // Add to pending actions collection
+    this._pendingActions.push(pending);
+
+    // Ensure action is removed when resolved
+    const remove = () => {
+      const i = this._pendingActions.indexOf(pending);
+      this._pendingActions.splice(i, 1);
+      this.stateChanged.emit(undefined);
+    };
+    pending.then(remove, remove);
+
+    // Signal changed state
+    this.stateChanged.emit(undefined);
+  }
+
+  /**
+   * Refresh plugin lock statuses
+   */
+  async refresh(): Promise<void> {
+    this.statusError = null;
+    this._isLoading = true;
+    this.stateChanged.emit();
+    try {
+      // Get the lock status from backend; if backend is not available,
+      // we assume that all plugins are locked.
+      const fallback: IPluginManagerStatus = {
+        allLocked: true,
+        lockRules: []
+      };
```

Comment:
@jtpio here we allow the endpoint to be unavailable, the view/search functionality is still available but all plugins are shown as locked (so not possible to enable/disable). Is that what you were asking about?

Reply:
Nice. Yes the question was mostly about being able to use the table view in cases where the backend is not available.